sdk/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/Hash.scala

```diff
@@ -174,6 +230,15 @@ object Hash {
       add(bytes.length).add(bytes)
     }
 
+    private def addRecord(fs: ImmArray[(_, Value)]): this.type =
+      if (upgradable) {
+        fs.iterator.zipWithIndex.foreach { case ((_, v), i) =>
+          discard(addTypedValue(i, v))
+        }
```

Comment:
First example does not work because we do not have char in Daml, but the second one does.
Good catch.

Reply:
I added two collision tests for the two cases you show up (Since there is no char type in Daml, I needed to adapt a bit the first case).
See 74f65f371

Reply:
In fact the issue is not the Record but the list/map.
`List("", "a")` clashes with `List("a", "")`

Reply:
> I was able to simplify even further your second case `List({})` clashes with `List({1:"a"})`

I'm not following. Do you mean `List({1:""})`? Otherwise I don't see how the non-empty string `"a"` can disappear from the hash.

Reply:
my bad